src/main/java/org/embulk/input/mongodb/MongodbInputPlugin.java

```diff
@@ -175,10 +165,19 @@ public TaskReport run(TaskSource taskSource,
         }
 
         if (task.getAggregation().isPresent()) {
-            Bson aggregationString = Document.parse(task.getAggregation().get());
-            List<Bson> aggregation = Arrays.asList(aggregationString);
+            Object jsonValue = Document.parse("{\"json\":" + task.getAggregation().get() + "}").get("json");
```

Comment:
This is a simple technique to parse either an array or a document.

Reply:
> Is it really safe, for example, against a kind of "JSON injection"? I'd choose a type of foolishly naive way rather than this type of quick string techniques.

The code uses Mongo libraries to parse JSON, and I couldn't find a generic method to parse an array _or_ a document. A better solution is to use a JSON processing library like Jackson instead. The approach I put in avoids the problem by turning the array into a subfield of a document, letting the parsing deal with documents.

I don't see any new meaningful attack here, any attack would have been possible before. Plus the parse function will break on invalid JSON.